detector/tracker/DriftChamber_o1_v02.cpp

```diff
@@ -510,6 +515,81 @@ static dd4hep::Ref_t create_DCH_o1_v02(dd4hep::Detector &desc, dd4hep::xml::Hand
 
 }
 
+inline double Circumradius(double Apothem, double dphi){return Apothem/cos(0.5*dphi/dd4hep::rad);}
+
+dd4hep::Solid CompositeTT(double twist_angle,  double cell_rin_z0,  double cell_rout_z0, double dz, double dphi, const dd4hep::rec::DCH_info & DCH_i)
+{
+
+  //----------------- G. trapezoid -------------
+  // make generic trapezoid bigger, later intersected with hyperboloid of proper radii
+  double rmin_zLhalf = DCH_i.Radius_zLhalf(cell_rin_z0 );
+  double rout_zLhalf = DCH_i.Radius_zLhalf(cell_rout_z0);
+  double trap_rin    = 0.9*rmin_zLhalf;
+  double trap_rout   = Circumradius(1.1*rout_zLhalf, dphi);
+
+  double poly_angle = dphi/2;
+  double twist_angle_half = twist_angle/2.;
+  // change sign, so the final shape has the same orientation as G4 twisted tube
+  twist_angle_half*= -1;
+
+  // define points of 8 genenric trapezoid
+  struct point2d
+  {
+    double x = {0.0};
+    double y = {0.0};
+  };
+
+  struct face
+  {
+    point2d A;
+    point2d B;
+    point2d C;
+    point2d D;
+  };
+
+  face fZneg;
+  face fZpos;
+
+  fZpos.A = { trap_rin*cos(  poly_angle + twist_angle_half ), trap_rin*sin(  poly_angle + twist_angle_half ) };
```

Comment:
can we get some general comments what this is doing? Why it is doing it?

Reply:
I have added the following comments, is it enough? I think it is challenging to explain without a picture
```
/// Solid equivalent to a twisted tube, resulting from the intersection of an hyperboloid and a generic trapezoid
/// the hyperboloid provides the hyperboloidal surfaces, the trapezoid provides the other two types of surfaces
```

Reply:
I am writing a technical note with plots and equations, I think it is better suited place to explain what is happening in that piece of code. I can add later a reference to the note once it is uploaded to CDS